app/uk/gov/hmrc/cdsreimbursementclaimfrontend/connectors/UpscanConnector.scala

```diff
@@ -65,38 +66,32 @@ class DefaultUpscanConnector @Inject() (
 ) extends UpscanConnector
     with Logging {
 
-  private def getUpscanInitiateConfig[A : ConfigReader](key: String): A =
-    config.underlying
-      .get[A](s"microservice.services.upscan-initiate.$key")
-      .value
-
   private val upscanInitiateUrl: String = {
-    val protocol = getUpscanInitiateConfig[String]("protocol")
-    val host     = getUpscanInitiateConfig[String]("host")
-    val port     = getUpscanInitiateConfig[String]("port")
+    val protocol = config.readUpscanInitServiceProtocol
+    val host     = config.readUpscanInitServiceHost
+    val port     = config.readUpscanInitServicePort
     s"$protocol://$host:$port/upscan/v2/initiate"
   }
 
   private val baseUrl: String = servicesConfig.baseUrl("cds-reimbursement-claim")
 
-  private val selfBaseUrl: String = config.underlying.get[String]("self.url").value
-
-  private val maxFileSize: Long = getUpscanInitiateConfig[Long]("max-file-size")
-
-  override def initiate(
+  override def initiate[A](
     errorRedirect: Call,
     successRedirect: Call,
     uploadReference: UploadReference
   )(implicit
-    hc: HeaderCarrier
+    hc: HeaderCarrier,
+    fileUpload: FileUpload[A]
   ): EitherT[Future, Error, HttpResponse] = {
 
+    val selfBaseUrl = config.readSelfBaseUrl
+
     val payload = UpscanInitiateRequest(
       baseUrl + s"/cds-reimbursement-claim/upscan-call-back/upload-reference/${uploadReference.value}",
       selfBaseUrl + successRedirect.url,
       selfBaseUrl + errorRedirect.url,
       0,
-      maxFileSize
+      config.readMaxFileSize(fileUpload.key)
     )
 
     logger.info(
```

Comment:
This log call needs to have additional information to tell us which type of upload we are doing. It would be sufficient just to take the name of type `A` and add it to the log message so we can debug stuff in Production easier.